### PR TITLE
#194 - Authorization header parsing extracted

### LINKED ISSUE #194: Authorization.java:34-37: Extract header value parsing...
The puzzle `191-868754ef` from #191 has to be resolved: 

https://github.com/artipie/http/blob/eec2ba77a2ccc115df8014f7c0b44bfacf4614b9/src/main/java/com/artipie/http/headers/Authorization.java#L34-L37

The puzzle was created by @olegmoz on 02-Jul-20. 

Estimate: 60 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #194
Authorization header parsing extracted

### CODE REVIEW

File: src/main/java/com/artipie/http/headers/Authorization.java [hunk 1]
```diff
@@ -73,6 +77,40 @@ public Authorization(final Headers headers) {
...
+
+    /**
+     * Creates matcher for header value.
+     *
+     * @return Matcher for header value.
+     */
+    private Matcher matcher() {
+        final String value = this.getValue();
+        final Matcher matcher = VALUE.matcher(value);
+        if (!matcher.matches()) {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/http/headers/Authorization.java:106]: @olegmoz codecov is complaining that this branch is not covered with test, I'd suggest to cover it

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz one comment, take a look please

**Comment by olegmoz** [comment]
@olenagerasimova thanks, adding tests you suggested I understood that RegEx for header value was too lax. Please review

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, good to merge

**Pull request merged**